stardis/radiation_field/radiation_field_solvers/base.py

```diff
@@ -152,15 +296,28 @@ def raytrace(stellar_model, stellar_radiation_field, no_of_thetas=20):
     thetas = np.linspace(start_theta, end_theta, no_of_thetas)
 
     ###TODO: Thetas should probably be held by the model? Then can be passed in from there.
-    for theta in thetas:
-        weight = 2 * np.pi * dtheta * np.sin(theta) * np.cos(theta)
-        stellar_radiation_field.F_nu += weight * single_theta_trace(
-            stellar_model.geometry.dist_to_next_depth_point,
-            stellar_model.temperatures.value.reshape(-1, 1),
-            stellar_radiation_field.opacities.total_alphas,
-            stellar_radiation_field.frequencies,
-            theta,
-            stellar_radiation_field.source_function,
-        )
+    if parallel_config is False:
+        for theta in thetas:
+            weight = 2 * np.pi * dtheta * np.sin(theta) * np.cos(theta)
+            stellar_radiation_field.F_nu += weight * single_theta_trace(
+                stellar_model.geometry.dist_to_next_depth_point,
+                stellar_model.temperatures.value.reshape(-1, 1),
+                stellar_radiation_field.opacities.total_alphas,
+                stellar_radiation_field.frequencies,
+                theta,
+                stellar_radiation_field.source_function,
+            )
+
+    elif parallel_config:
+        for theta in thetas:
+            weight = 2 * np.pi * dtheta * np.sin(theta) * np.cos(theta)
```

Comment:
If you're already doing a lot of loop unraveling, this may as well be calculated outside the loop

Reply:
I've done this for the single threaded case where I'm letting numpy do the calculations, but I don't think this works for the lower level njit code.

Reply:
The benchmarks report that that change slows down the function by another .1 seconds which seems quite large. Not sure if that persists when running a larger simulation or not.